### DISCUSSION TITLE
Replacement for the old '/vehicles/{VIN}/executeService' endpoint?

### DISCUSSION THREAD

**Discussion by Nhubbers** [bimmerconnected/bimmer_connected#387] [Feature request]
Love the work done here! 

Besides collecting data, I previously used the API's to schedule the charging process remotely. 

I was using the following endpoint:
```/webapi/v1/user/vehicles/{VIN}/executeService?serviceType={COMMAND}``` 

To make the car's charging schedule change to immediate charging:
```?serviceType= CHARGE_NOW```

OR To update the Charge Schedule in the car, i used:
and: ```?serviceType=CHARGING_CONTROL```

for example in combination with a schedule like: 

```
{
    "serviceType": ["CHARGING_CONTROL"],
    "data": [
        """{
            'weeklyPlanner': {
                'climatizationEnabled': True,
                'chargingMode': 'DELAYED_CHARGING',
                'chargingPreferences': 'CHARGING_WINDOW',
                'timer1': {
                    'departureTime': '07:00',
                    'timerEnabled': True,
                    'weekdays': [
                        'MONDAY',
                        'TUESDAY',
                        'WEDNESDAY',
                        'THURSDAY',
                        'FRIDAY'
            ],
                },
                'timer2': {
                    'departureTime': '8:00',
                    'timerEnabled': False,
                    'weekdays': ['SATURDAY','SUNDAY'],
                },
                'timer3': {
                    'departureTime': '08:00',
                    'timerEnabled': False,
                    'weekdays': [],
                },
                'overrideTimer': {
                    'departureTime': '07:30',
                    'timerEnabled': False,
                    'weekdays': ['MONDAY'],
                },
                'preferredChargingWindow': {
                    'enabled': True,
                    'startTime': '23:00',
                    'endTime': '07:00',
                },
            }
        }"""
    ],
}
```

After excecution i collected the status of the command through: 

```/webapi/v1/user/vehicles/{VIN}/serviceExecutionStatus?serviceType={SERVICE}```

**In the new API I haven't been able to find the new route to this service. 
Anyone who has found an answer?**

**Comment by rikroe** [comment]
The routes should still exist, please have a look at [bimmer_connected/const.py#L15-L18](https://github.com/bimmerconnected/bimmer_connected/blob/master/bimmer_connected/const.py#L15-L18) for the URL templates.

We haven't implemented the `CHARGE_NOW` or `CHARGING_CONTROL` services yet. While the first one should be easy, I'm not so sure about the second, as you need in depth-knowledge of the structure that the BMW API is expecting.

**Reply by jertje260** [reply]
I saw this, and I gave it a try, however it returns a 502 Bad Gateway error for me. Any ideas on how to get this to work?
